File: backend/app/schemas/school.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr


class SchoolBase(BaseModel):
    name: str
    email: EmailStr
    phone: Optional[str] = None
    address: Optional[str] = None
    state: Optional[str] = None
    country: Optional[str] = None
    motto: Optional[str] = None
    website: Optional[str] = None
    principal_name: Optional[str] = None
    primary_color: Optional[str] = None
    secondary_color: Optional[str] = None


class SchoolCreate(SchoolBase):
    pass


class SchoolRead(SchoolBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    is_active: bool
    logo_path: Optional[str] = None
    signature_path: Optional[str] = None
    stamp_path: Optional[str] = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/school.ts
export interface School {
  id: number;
  name: string;
  email: string;
  phone: string | null;
  address: string | null;
  state: string | null;
  country: string | null;
  motto: string | null;
  website: string | null;
  principal_name: string | null;
  primary_color: string | null;
  secondary_color: string | null;
  is_active: boolean;
  logo_path: string | null;
  signature_path: string | null;
  stamp_path: string | null;
  created_at: string;
  updated_at: string;
}
